apps/extension/client/src/checkTypegenNesting.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from "vscode";
+
+const typeGenPatternKey = "*.ts";
+const typeGenPattern = "${capture}.typegen.ts";
+const hasTypeGenPattern = (fileNestingPatterns: object) =>
```

Comment:
we are currently only checking workspace settings, I wonder if we should respect user setting here too - although I don't have strong recommendations as to how this should work. Pehaps we should check both for the pattern but keep writing to workspace-level settings?

Reply:
we check the full config now since we're not providing a scope when we get the configuration.